indexing-hadoop/src/main/java/io/druid/indexer/HadoopDruidIndexerMapper.java

```diff
@@ -103,7 +103,9 @@ public final static InputRow parseInputRow(Object value, InputRowParser parser)
   {
     if(parser instanceof StringInputRowParser && value instanceof Text) {
       //Note: This is to ensure backward compatibility with 0.7.0 and before
-      return ((StringInputRowParser)parser).parse(value.toString());
+      return ((StringInputRowParser) parser).parse(value.toString());
+    } else if(value instanceof InputRow) {
+      return (InputRow)value;
```

Comment:
I haven't full thought this through, but is there any harm in having the SegmentInputFormat know about the Aggregators such that it can return the "name" value when it is asked for "fieldName" as well?